### PR TITLE
feat(doc): warn when replace_range targets a heading but markdown lacks matching heading

### PR DESCRIPTION
**Pull request by herbertliu**
## Summary

Addresses Case 2 in the lark-cli pitfall list: `replace_range` cannot change a block's type, so using it to "demote" a heading block to a paragraph silently fails — the API returns success but the heading styling persists.

This PR extends the existing `docsUpdateWarnings` infrastructure (from the PR #569 pre-write semantic check series) with a targeted, high-signal warning: when the invocation is `--mode=replace_range --selection-by-title "## X"` and `--markdown` does not lead with a matching heading level, the user almost certainly expects a block-type change that won't happen.

## Changes

- **`shortcuts/doc/docs_update_check.go`**:
  - New `checkDocsUpdateReplaceHeadingBlockType` — fires only on `replace_range` + heading-style `--selection-by-title` + non-matching first-prose-line heading level
  - New narrow helpers `atxHeadingLevel` (ATX-only, CommonMark-conformant) and `firstProseHeadingLevel` (reuses existing fence-tracking helpers)
  - Signature of `docsUpdateWarnings` extended with `selectionByTitle` argument
- **`shortcuts/doc/docs_update.go`**: pass `--selection-by-title` through to `docsUpdateWarnings`
- **`shortcuts/doc/docs_update_check_test.go`**: table-driven tests (8 cases on the check + 9 on the ATX helper); 100% coverage on the new function

## Scope notes

The warning is deliberately narrow:

- Fires only when `--selection-by-title` is used. Other selection modes (`--selection-with-ellipsis`, bare token) can target any block and would produce false positives.
- Fires only when the markdown's first non-blank, non-fenced prose line is not an ATX heading of the same level.
- Non-blocking — the update still proceeds, consistent with the existing warnings from #569.

## Test Plan

- [x] `go test ./shortcuts/doc/...` passes
- [x] `go vet ./...` clean
- [x] `gofmt -l` clean
- [x] New coverage on `checkDocsUpdateReplaceHeadingBlockType` is 100%

## Summary by CodeRabbit

* **Bug Fixes**
  * Added a validation warning when using replace mode with a title-based selection: alerts if the selected heading style/level conflicts with the replacement and recommends delete + insert instead.

* **Tests**
  * Expanded test coverage for heading detection and warning generation, including ATX heading edge cases and skipped fenced-code scenarios.

### CODE REVIEW

File: shortcuts/doc/docs_update_check.go [hunk 1]
```diff
@@ -34,17 +34,103 @@ import (
...
+	if level == 0 || level > 6 {
+		return 0
+	}
+	if level == len(trimmed) {
+		return 0
+	}
+	if trimmed[level] != ' ' {
+		return 0
+	}
+	return level
```
[the remark this answers is not in the document]
Comment (herbertliu) [reply shortcuts/doc/docs_update_check.go]: Fixed in 13829fa — atxHeadingLevel now uses fenceIndentOK for CommonMark 0..3 leading-space semantics, accepts space/tab/EOL after the hash run, and treats empty headings (e.g. '##') as valid. Tests updated with the 4 cases you called out ('    ## code':0, '\t## code':0, '##\tH2':2, '##':2).

File: shortcuts/doc/docs_update_check.go [hunk 2]
```diff
@@ -34,17 +34,117 @@ import (
...
+		return 0
+	}
+	if level == len(body) {
+		return level
+	}
+	switch body[level] {
+	case ' ', '\t':
+		return level
+	}
+	return 0
```
[the remark this answers is not in the document]
Comment (herbertliu) [reply shortcuts/doc/docs_update_check.go:115]: Addressed together with the atxHeadingLevel fix in 13829fa; the two comments share the same root cause.

**Comment by herbertliu** [comment]
Closing this PR — superseded by the docs v2 API refactor (#638).

v2 introduces `--command block_replace` (and other block-level commands) that uses a precise `--block-id` for targeting. Block-type changes are now expressed by `block_delete` + `block_insert_after`, eliminating the entire 'replace_range cannot change block type' class of bugs that this warning was designed to catch.

The v1 `--mode` selection-by-title path still exists for backward compat but is hidden/deprecated. Adding a warning to a deprecated path provides limited value going forward.

If anyone is still on v1 long enough to need this warning, the v1 deprecation message itself is the better signal.
